client/signup/config/flows.js

```diff
@@ -157,13 +159,16 @@ function removeUserStepFromFlow( flow ) {
 	} );
 }
 
-function getCurrentFlowNameFromTest() {
-	// No tests currently running
+function getCurrentFlowName( currentUrl ) {
+	// Headstart test - Only consider users from the general /start path.
+	if ( '/start' === currentUrl && 'headstart' === abtest( 'headstart' ) ) {
```

Comment:
On production, users are usually coming from the homepage, where the signup link is actually `/start/en?ref=homepage` (for EN users).